Modules/_winapi.c

```diff
@@ -112,12 +112,22 @@ typedef struct {
     Py_buffer write_buffer;
 } OverlappedObject;
 
+static int
+overlapped_traverse(OverlappedObject *self, visitproc visit, void *arg)
+{
+    Py_VISIT(self->read_buffer);
+    Py_VISIT(self->write_buffer.obj);
+    Py_VISIT(Py_TYPE(self));
+    return 0;
+}
+
 static void
 overlapped_dealloc(OverlappedObject *self)
 {
     DWORD bytes;
     int err = GetLastError();
 
+    PyObject_GC_UnTrack(self);
```

Comment:
What happens if you un-track, enter the `if` below, and return in line 157? Would it be better to un-track after the `if`? I don't know the answer :)

Reply:
Thanks for pointing that out. That's a pretty scary thought. I moved it down to somewhere much safer (after all the conditionals).